### PR TITLE
Fix keyCompromise behavior

### LINKED ISSUE #208: Clarify when keyCompromise revocation results in the key being blocked
When a subscriber requests revocation for reason keyCompromise, but has _not_ proved possession of that key, we accept the revocation request but do not block future issuance for that key. When a revocation request is signed with the certificate key itself, we use reason keyCompromise and block the key regardless of the reason included in the request.

We should update the text in Section 4.9.3 to clarify this.

### PR DESCRIPTION
**Pull request by andygabby**
Modify key compromise behavior to reflect current
practice and intent.

Fixes #208

### CODE REVIEW

File: CP-CPS.md [hunk 1]
```diff
@@ -473,7 +473,11 @@ ISRG may administratively revoke certificates if it determines that the Subscrib
...
+Anyone may make revocation requests, at any time, via the Certificate Revocation interface of the ACME Protocol defined in RFC 8555 section 7.6.
+
+A successful revocation request signed by the private key corresponding to the public key in the certificate, per the requirements in Section 4.9.12 of this document, will result in the certificate being revoked with reason code `keyCompromise` and the affected key being blocked for future issuance. All currently valid certificates with that key will also be revoked.
+
+A successful revocation request where a subscriber has demonstrated control over all domains in a certificate and has requested revocation with reason code `keyCompromise` may result in the certificate being revoked with reason code `keyCompromise` but will not result in blocking the key for future issuance or revoking other certificates with the same key unless proof of possession of the private key is demonstrated per the requirements in Section 4.9.12 of this document.
```
Comment (beautifulentropy) [inline CP-CPS.md:480]: The paragraph above this feels really nice and concise. I've reworked this one to feel about the same. Feel free to disregard.

```suggestion
If a subscriber has demonstrated control over all domains in a certificate and requests revocation citing `keyCompromise`, the certificate may, at our discretion, be revoked with this reason code. However, this action won't block the key for future issuance nor revoke other certificates using the same key, unless proof of possession of the private key is provided as outlined in Section 4.9.12.
```

File: CP-CPS.md [hunk 2]
```diff
@@ -520,7 +524,9 @@ ISRG allows for OCSP stapling.
 
 ### 4.9.12 Special requirements re key compromise
 
-Key compromise must be demonstrated via the Certificate Revocation method of the ACME Protocol defined in RFC 8555, Section 7.6 by signing the request using the private key corresponding to the public key in the certificate (not the private key corresponding to the account which originally requested the certificate).
+In order for a successful revocation request to result in the blocking of a key for future issuance and revocation of all currently valid certificates with that key, proof of possession of the private key for the certificate must be demonstrated.
+
+Proof of possession of a private key corresponding to the public key of a certificate must be demonstrated via the Certificate Revocation method of the ACME Protocol defined in RFC 8555, Section 7.6 by signing the request using the private key corresponding to the public key in the certificate (not the private key corresponding to the account which originally requested the certificate).
```
Comment (beautifulentropy) [inline CP-CPS.md:529]: A version of this which should convey all the same information but is more in keeping with the brevity of the original.

```suggestion
To block a key for future issuance and revoke all valid certificates associated with that key, the requester must demonstrate possession of the certificate’s private key. This proof must be provided via the Certificate Revocation method in the ACME Protocol, as defined in RFC 8555, Section 7.6. The revocation request must be signed using the private key that corresponds to the certificate's public key. This key must be distinct from the private key associated with the account that originally requested the certificate.
```

**Review by beautifulentropy — COMMENTED** [review]
Some suggestions.

File: CP-CPS.md (hunk 1 above)
Comment (ezekiel) [reply CP-CPS.md:480]: It does seem important to maintain `a successful revocation request`.

**Review by jsha — APPROVED** [review]

File: CP-CPS.md [hunk 3]
```diff
@@ -520,7 +520,11 @@ ISRG allows for OCSP stapling.
 
 ### 4.9.12 Special requirements re key compromise
 
-Key compromise must be demonstrated via the Certificate Revocation method of the ACME Protocol defined in RFC 8555, Section 7.6 by signing the request using the private key corresponding to the public key in the certificate (not the private key corresponding to the account which originally requested the certificate).
+ISRG considers key compromise demonstrated when revocation is successfully requested via the ACME Protocol and the request was signed by the private key of the certificate. Key compromise can be demonstrated via ACME even for certificates that have previously been revoked without demonstrating key compromise.
+
+ISRG may also consider key compromise demonstrated by non-ACME methods.
+
+When key compromise is demonstrated, ISRG blocks the key from use in future issuance and revokes all certificates that used that key.
```
Comment (jcjones) [inline CP-CPS.md:527]:
```suggestion
When key compromise is demonstrated, ISRG blocks the key from use in future issuance and revokes all unexpiredcertificates that used that key.
```

**Review by mcpherrinm — APPROVED** [review]

**Review by bdaehlie — APPROVED** [review]

**Pull request merged**